andela/rivendell-ah
# #159204691: user should receive a json web token on authentication

#### What does this PR do?
- adds  signup test to the app
- adds a service folder to the app
- creates a .env-sample file to the app

#### Description of Task to be completed?
- as a user, I should receive a JSON Web Token on authentication

#### How should this be manually tested?
- run the command `npm install` to install all the dependencies
- run `npm start` in order to start the server
- create a post request to **localhost:3000/api/users/**. The body of this request should be similar to 
have the properties in the JSON below:
{
	"user" : {
		"email": "chidi@email.com",
		"username": "username101",
		"password": "password1"
	}
}
- you should receive a response like this:

{
    "data": {
        "email": "chidi@email.com",
        "username": "username101",
        "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbCI6ImNoaWRpQGVtYWlsLmNvbSIsInVzZXJuYW1lIjoidXNlcm5hbWUxMDEiLCJwYXNzd29yZCI6InBhc3N3b3JkMSIsImlhdCI6MTUzMjQ1NDE0NCwiZXhwIjoxNTMyNzEzMzQ0fQ.njMCExUNOhkQwLk4vzE9ktCQvG1LBaWqNXfCnpprumw"
    }
}


#### Any background context you want to provide?
- Ensure that mongodb is set up and running
#### What are the relevant pivotal tracker stories?
- 159204691
#### Screenshots (if appropriate):
<img width="1146" alt="screen shot 2018-07-24 at 6 45 28 pm" src="https://user-images.githubusercontent.com/38565349/43156353-cdd1683a-8f71-11e8-88f5-ccd461815508.png">

#### Questions:
- None
